### ISSUE TITLE
Visual beep doesn't work on OSX

### ISSUE THREAD

**Issue by benetherington** [KSP-KOS/KOS#1352]
On OS 10.11.2 installed on my 2014 MBP, visual beep doesn't work. The button toggles correctly with key presses and `terminal:visualbeep`, but the actual beep behavior isn't affected.

**Comment by Dunbaratu** [comment]
So does it fail to flash the terminal, or does it still make a sound, or both?  I could believe that it fails to flash, or still makes a sound, but I can't imagine how the OS makes any difference at all on whether or not the setting is used.

**Comment by benetherington** [comment]
No flash, still makes a sound.

You're right, I doubt the OS makes a difference, but I can't think of any other difference. I'm running in a clean install with Kerbal Engineer and no other mods. Let me try again in a completely fresh install.

**Comment by Dunbaratu** [comment]
Also check the output log for any exceptions that might not have been reported on screen.  They can cause code to be aborted before it hits the relevant parts.

**Comment by benetherington** [comment]
Confirmed in a brand new install of KSP 1.0.5 and KOS 0.18.2. As above, the button toggles but there's no flash and no beep. The invert screen button works as expected.

I don't see any additional exceptions reported in [the log](https://github.com/KSP-KOS/KOS/files/81929/KSP.log.txt), either.

**Comment by Dunbaratu** [comment]
Then I just don't understand, because it works when I try it.  I can't see anything that would differ, although I'm not trying it on a Mac, I can't see anything in the code that should be OS-dependant in terms of using that toggle flag.

I mean it's just this:
```
        bool Beep()
        {
            if (shared.Screen.VisualBeep)
            {
                DateTime nowTime = DateTime.Now;
                if (nowTime < (blinkEndTime + (blinkCoolDownDuration))) // prev blink not done yet.
                    return false;
                
                // Turning this timer on tells GUI repainter elsewhere in this class to paint in reverse until it expires:
                blinkEndTime = nowTime + blinkDuration;
            }
            else
            {
                if (!beepSource.clip.isReadyToPlay || beepSource.isPlaying)
                    return false; // prev beep sound still is happening.
                
                // This is nonblocking.  Begins playing sound in background.  Code will not wait for it to finish:
                beepSource.Play();
            }
            return true;
        }
```
If the visualbeep flag is on, it should be literally impossible to ever reach the else condition that tries to play the sound.

**Comment by hvacengi** [comment]
I happened to be in the Logger looking at something else, and suddenly had a thought: @benetherington you do mean the beep that currently requires KSlib’s spec_char.ksm file, or are you referring to the error beep when a script throws an exception?  I noticed that you had 2 exceptions thrown at the end of the posted log file, which made me wonder if that was what you were referring too.  The error sound doesn't get replace by the visual beep setting.

**Comment by Dunbaratu** [comment]
Hmm that's a good point.  I hadn't thought of that.  The error beep is meant to get your attention even when the terminal isn't showing so if you're someone who leaves the terminal closed and communicates only with `hudtext` to the user, you still get an indication.   So it's not allowed to be silenced.  Maybe we could change it to throw a `hudtext` message on the screen if you have visual beep off and the terminal is closed.

**Comment by hvacengi** [comment]
I've noticed that hudtext will only display one instance of a given string at any given time.  If you resend the same string it just appears to restart the timeout clock.  We could use that to our advantage, and continue to send the error hudtext until the terminal is displayed.  That would be very similar to your previous suggestion of changing the screen color until the terminal is focused.

**Comment by benetherington** [comment]
Given what @hvacengi said, I must be misunderstanding the way visual beep works. I've been talking about the exception beep. It drives me nuts, because I rarely write code that isn't missing a period SOMEWHERE.

So, yeah, can we make this a feature request along the lines of the above?
